core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -986,7 +986,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     union(Seq(first) ++ rest)
 
   /** Get an RDD that has no partitions or elements. */
-  def emptyRDD[T: ClassTag] = new EmptyRDD[T](this)
+  def emptyRDD[T: ClassTag]: EmptyRDD[T] = new EmptyRDD[T](this)
```

Comment:
shouldn't the return type here by `RDD[T]`, since `EmptyRDD` is `private[spark]` and just an implementation detail?